.gitignore

```diff
@@ -1,2 +1,5 @@
 /target/
```

Comment:
We should add in the eclipse generated files like .project, .classpath, and .settings to the .gitignore.

Reply:
We should also remove all these eclipse files from the PR